### PR TITLE
refactor(shorebird_cli): delete old patch command and mixins

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Removes old patch command now that new command has been introduced.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

**Review by felangel — APPROVED** [review]
🔥

File: packages/shorebird_cli/lib/src/commands/patch/patch_command.dart [hunk 1]
```diff
@@ -82,7 +82,7 @@ of the iOS app that is using this module.''',
       )
       ..addFlag(
         'codesign',
-        help: 'Codesign the application bundle.',
+        help: 'Codesign the application bundle (iOS only).',
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_command.dart:85]: Maybe we should add `(iOS only)` to the exportOptionsPlist as well?

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_command.dart:85]: Done

Comment (eseidel) [reply packages/shorebird_cli/lib/src/commands/patch/patch_command.dart:85]: I suspect we will eventually end up wtih a whole section of iOS stuff, Android stuff, etc.  But marking them for now sgtm.

**Pull request merged**